### PR TITLE
python3Packages.ovito: 3.0.0 -> 3.3.1

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

Version update, convert to Python 3 (part of https://github.com/NixOS/nixpkgs/issues/101964), fix build

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ovito/default.nix [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
+, netcdf
+, qscintilla
+, zlib
+, boost
+, git
+, fftw
+, hdf5
+, libssh
+, qt5
+, python3Packages
```
Comment (jonringer) [inline pkgs/development/python-modules/ovito/default.nix:14]:
```suggestion
, python
```

File: pkgs/development/python-modules/ovito/default.nix [hunk 2]
```diff
@@ -1,34 +1,43 @@
...
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "0rm1qxa0fanaaqg0idr6rf2s2xlbyn1dzjzwh3rddy9mgl60lj2h";
   };
 
-  buildInputs = [ cmake libav netcdf qscintilla zlib boost zlib git fftw hdf5 libssh ];
-  propagatedBuildInputs = with pythonPackages; [ sphinx numpy sip pyqt5 matplotlib ase ];
+  buildInputs = [ cmake ffmpeg netcdf qscintilla zlib boost zlib git fftw hdf5 libssh qt5.qtbase qt5.qtsvg ];
+
+  propagatedBuildInputs = with python3Packages; [ sphinx numpy sip pyqt5 matplotlib ase ];
```
Comment (jonringer) [inline pkgs/development/python-modules/ovito/default.nix:30]:
```suggestion
  propagatedBuildInputs = with python.pkgs; [ sphinx numpy sip pyqt5 matplotlib ase ];
```

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -4272,7 +4272,7 @@ in {
 
   ovh = callPackage ../development/python-modules/ovh { };
 
-  ovito = toPythonModule (pkgs.libsForQt5.callPackage ../development/python-modules/ovito { pythonPackages = self; });
+  ovito = toPythonModule (pkgs.libsForQt5.callPackage ../development/python-modules/ovito { python3Packages = self; });
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:4275]:
```suggestion
  ovito = toPythonModule (pkgs.libsForQt5.callPackage ../development/python-modules/ovito { });
```

**Review by jonringer — COMMENTED** [review]
I would much rather have the python interpreter parameterized

in all-packages, i would would just do `python = python3;`

**Comment by jonringer** [comment]
do you mind squashing? 

Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

This can be done without `git rebase -i` by doing:
```
git reset HEAD~1                    # move fix-up commits into unstaged
git add -- pkgs/                    # move changes into staged
git commit --amend --no-edit        # add changes to previous commit
git push ... ... --force            # modify current PR branch
```

However, `git rebase -i` is a more powerful alternative, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Comment by prusnak** [comment]
> do you mind squashing?

I am squashing it at the moment. Easiest thing how to accept your suggestions in my local branch is to accept them via Github UI into remote branch and then update my local branch.

**Comment by prusnak** [comment]
> in all-packages, i would would just do `python = python3;`

What do you mean by this? This changeset does not touch all-packages ...

**Comment by jonringer** [comment]
sorry, for some reason I thought there was a top-level ovito

**Comment by jonringer** [comment]
oh in the original list, it just shows `ovito`, but that's likely because it's just taking the derivation name.

File: pkgs/development/python-modules/ovito/default.nix [hunk 4]
```diff
@@ -1,34 +1,43 @@
...
+  version = "3.3.1";
   pname = "ovito";
 
-  src = fetchgit {
-    url = "https://gitlab.com/stuko/ovito";
-    rev = "a28c28182a879d2a1b511ec56f9845306dd8a4db";
-    sha256 = "1vqzv3gzwf8r0g05a7fj8hdyvnzq2h3wdfck7j6n1av6rvp7hi5r";
+  src = fetchFromGitLab {
+    owner = "stuko";
+    repo = "${pname}";
```
Comment (jonringer) [inline pkgs/development/python-modules/ovito/default.nix:23]: but this is minor
```suggestion
    repo = pname;
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 102250 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>python27Packages.ovito</li>    <li>python37Packages.ovito</li>    <li>python38Packages.ovito</li>  </ul></details>

File: pkgs/development/python-modules/ovito/default.nix (hunk 4 above)
Comment (prusnak) [reply pkgs/development/python-modules/ovito/default.nix:23]: Amended in 46a8d9d82ac51569fefbc4001e832bc799b7ef05

**Pull request merged**